### PR TITLE
feat: Add `avoid_single_child` rule

### PR DESCRIPTION
**Pull request by naipaka**
## 🔗 Related Issues

- None

## 🙌 What's Done

- Add `avoid_single_child` rule

## ✍️ What's Not Done

- None

## 🖼️ Image Differences

CleanShot 2024-07-29 at 12 32 28

## 🤼 Desired Review Method

- [ ] Correction Commit
- [ ] Pair programming

> [!NOTE]
> It is possible that a reviewer's will may cause a method to be implemented that is not selected.

## Pre-launch Checklist
- [x] I have reviewed my own code.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I updated/added relevant documentation (doc comments with ///).

### CODE REVIEW

File: packages/altive_lints/lint_test/lints/avoid_shrink_wrap_in_list_view.dart [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// ignore_for_file: avoid_single_child
```
Comment (naipaka) [inline packages/altive_lints/lint_test/lints/avoid_shrink_wrap_in_list_view.dart:1]: I have added ignore comments to suppress warnings from tests that are not related to the rules added this time.

**Review by riscait — APPROVED** [review]
LGTM
We can prevent forgetting to turn it off!
